osquery/tables/system/windows/security_profile_info.cpp

```diff
@@ -0,0 +1,495 @@
+   */
+  HMODULE handleSceDLL_{nullptr};
+
+  /**
+   * @brief This function pointer points to exported SceFreeMemory in
+   * memory.
+   */
+  SceFreeMemoryPtr sceFreeMemory_{nullptr};
+
+  /**
+   * @brief This function pointer points to exported SceGetSecurityProfileInfo
+   * in memory.
+   */
+  GetSecProfileInfoFnPtr sceGetSecurityProfileInfo_{nullptr};
+
+  std::atomic<bool> initialized_{false};
+};
+
+SceClientHelper& SceClientHelper::instance() {
+  static SceClientHelper instance;
+  return instance;
+}
+
+SceClientHelper::SceClientHelper() {}
+
+SceClientHelper::~SceClientHelper() {
+  if (handleSceDLL_ != nullptr) {
+    if (!FreeLibrary(handleSceDLL_)) {
+      LOG(ERROR) << "Failed to free module handle of dll " << kTargetSCEDLL;
+    }
+
+    handleSceDLL_ = nullptr;
+  }
+
+  if (sceFreeMemory_ != nullptr) {
+    sceFreeMemory_ = nullptr;
+  }
+
+  if (sceGetSecurityProfileInfo_ != nullptr) {
+    sceGetSecurityProfileInfo_ = nullptr;
+  }
+
+  initialized_ = false;
+}
+
+Status SceClientHelper::initialize() {
+  // Checking first if the class is already initialized
+  if (initialized_) {
+    return Status::success();
+  }
+
+  // Checking if the input DLL is already mapped to memory before loading it.
+  // If mapped module is not found, LoadLibraryExA() gets called to load the
+  // module from system32 folder.
+  bool increasedRefCount = false;
+  HMODULE dllHandle = GetModuleHandleA(kTargetSCEDLL.c_str());
+  if (dllHandle == nullptr) {
+    // Library was not there in memory already, so we are loading here it and
+    // freeing it on the class destructor
+    increasedRefCount = true;
+    dllHandle = LoadLibraryExA(
+        kTargetSCEDLL.c_str(), NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
+  }
+
+  // An invalid module handle means that the DLL couldn't be loaded
+  if (dllHandle == nullptr) {
+    return Status::failure(kTargetSCEDLL + " dll couldn't be loaded");
+  }
+
+  // Getting the address to exported SceFreeMemory function
+  PVOID sceFreeMemoryAddr = GetProcAddress(dllHandle, kSceFreeMemoryFn.c_str());
+  if (sceFreeMemoryAddr == nullptr) {
+    if (increasedRefCount) {
+      FreeLibrary(dllHandle);
+    }
+    return Status::failure(kSceFreeMemoryFn + " couldn't be loaded");
+  }
+
+  // Getting the address to exported SceGetSecurityProfileInfo function
+  PVOID sceGetProfileInforAddr =
+      GetProcAddress(dllHandle, kSceGetSecProfileInfoFn.c_str());
+  if (sceGetProfileInforAddr == nullptr) {
+    if (increasedRefCount) {
+      FreeLibrary(dllHandle);
+    }
+    return Status::failure(kSceGetSecProfileInfoFn + " couldn't be loaded");
+  }
+
+  // Assigning the address of the exports in memory so they can be called thru
+  // function pointers that match the target function prototypes
+  sceFreeMemory_ = reinterpret_cast<SceFreeMemoryPtr>(sceFreeMemoryAddr);
+
+  sceGetSecurityProfileInfo_ =
+      reinterpret_cast<GetSecProfileInfoFnPtr>(sceGetProfileInforAddr);
```

Comment:
I'm sure I already know the answer here, but any chance a `static_cast` would work? We try and avoid `reinterpret_cast`s wherever possible, but I know often with the windows `PVOID` blobs there's not much option.

Reply:
Yes, using `static_cast` worked fine here. I tested it thoroughly, and also thought about it, and don't see any reason why it should not work. Thanks for the suggestion!. This was a nice catch indeed as I totally missed this one. I made the changes here and also addressed a C-like cast I had on table generation (I was using `(int)` there)